x-hooks/use-authentication.tsx

```diff
@@ -152,6 +154,10 @@ export function useAuthentication() {
           push(`/connect-account`);
           return false;
         }
+        if(user?.githubLogin){
```

Comment:
this shouldn't be needed, look for `changeCurrentUserMatch` and check why it's not working properly